src/client.js

```diff
@@ -221,6 +246,8 @@ class MistralClient {
    * @param {*} safePrompt whether to use safe mode, e.g. true
    * @param {*} toolChoice the tool to use, e.g. 'auto'
    * @param {*} responseFormat the format of the response, e.g. 'json_format'
+   * @param {*} [signal] - optional AbortSignal instance to control request
```

Comment:
Actually, it might be worth considering whether to move `signal` into a second object param, and keep it out of the bag of data that is sent to the server. It's a bit cleaner that way. 

On a related note, I realised that the JSDoc comments are actually incorrect ([see](https://github.com/mistralai/client-js/issues/70#issuecomment-2095044637))